Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs

```diff
@@ -204,89 +283,154 @@ protected void OnDestroy()
 
         #endregion MonoBehaviour Implementation
 
-        protected override void OnControllerFound()
-        {
-            if (transformTarget == null)
-            {
-                TrackTransform(ControllerTransform);
-            }
-        }
-
-        protected override void OnControllerLost()
-        {
-            DetachFromCurrentTrackedObject();
-        }
-
         /// <summary>
-        /// Clears the transform target and attaches to the current <see cref="TrackedObjectToReference"/>.
+        /// Clears the transform target and attaches to the current <see cref="TrackedTargetType"/>.
         /// </summary>
         public void RefreshTrackedObject()
         {
             DetachFromCurrentTrackedObject();
             AttachToNewTrackedObject();
         }
 
-        protected virtual void DetachFromCurrentTrackedObject()
+        public void SetTransformOverride(Transform target)
         {
-            transformTarget = null;
+            if (target != null)
+            {
+                this.transformOverride = target;
+                RefreshTrackedObject();
+            }
+        }
 
-            if (transformWithOffset != null)
+        protected virtual void DetachFromCurrentTrackedObject()
+        {
+            if (trackingTarget != null)
             {
-                Destroy(transformWithOffset);
-                transformWithOffset = null;
+                DestroyImmediate(trackingTarget);
+                trackingTarget = null;
             }
         }
 
         protected virtual void AttachToNewTrackedObject()
         {
-            switch (TrackedObjectToReference)
+            this.currentTrackedHandedness = Handedness.None;
+
+            Transform target = null;
+            if (TrackedTargetType == TrackedObjectType.Head)
+            {
+                target = CameraCache.Main.transform;
+            }
+            else if (TrackedTargetType == TrackedObjectType.MotionController)
+            {
+                if (this.TrackedHandness == Handedness.Both || this.TrackedHandness == Handedness.Any)
+                {
+                    this.currentTrackedHandedness = Handedness.Left;
+                    target = GetMotionController(Handedness.Left);
+                    if (target == null)
+                    {
+                        this.currentTrackedHandedness = Handedness.Right;
+                        target = GetMotionController(Handedness.Right);
+                        if (target == null)
+                        {
+                            this.currentTrackedHandedness = Handedness.None;
+                        }
+                    }
```

Comment:
Does this mean that if I have both controllers in the scene and I call RefreshTrackedObject, it will always prefer the left? Is there any way for the user to specify the handedness preference? I could just be misreading this :)

Reply:
Same comment on the hand joint stuff down below.

---
In reply to: [308996330](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5416#discussion_r308996330) [](ancestors = 308996330)

Reply:
It will always prefer Left if the TrackedHandedness value is set to Both. If not, then it will try to get the controller or hand of TrackedHandedness value (i.e right or left)